service/frontend/workflowHandler.go

```diff
@@ -1688,6 +1688,153 @@ func (wh *WorkflowHandler) StartWorkflowExecution(
 	return resp, nil
 }
 
+// GetRawHistory - retrieves raw history directly from DB layer
+func (wh *WorkflowHandler) GetWorkflowExecutionRawHistory(
+	ctx context.Context,
+	getRequest *gen.GetWorkflowExecutionRawHistoryRequest,
+) (resp *gen.GetWorkflowExecutionRawHistoryResponse, retError error) {
+	defer log.CapturePanic(wh.GetLogger(), &retError)
+
+	scope, sw := wh.startRequestProfileWithDomain(metrics.FrontendGetWorkflowExecutionRawHistoryScope, getRequest)
+	defer sw.Stop()
+
+	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+		return nil, wh.error(err, scope)
+	}
+
+	if getRequest == nil {
+		return nil, wh.error(errRequestNotSet, scope)
+	}
+
+	if ok := wh.allow(getRequest); !ok {
+		return nil, wh.error(createServiceBusyError(), scope)
+	}
+
+	if getRequest.GetDomain() == "" {
+		return nil, wh.error(errDomainNotSet, scope)
+	}
+
+	if err := wh.validateExecutionAndEmitMetrics(getRequest.Execution, scope); err != nil {
+		return nil, err
+	}
+
+	if getRequest.GetMaximumPageSize() <= 0 {
+		getRequest.MaximumPageSize = common.Int32Ptr(int32(wh.config.HistoryMaxPageSize(getRequest.GetDomain())))
+	}
+
+	domainID, err := wh.GetDomainCache().GetDomainID(getRequest.GetDomain())
+	if err != nil {
+		return nil, wh.error(err, scope)
+	}
+
+	// force limit page size if exceed
+	if getRequest.GetMaximumPageSize() > common.GetHistoryMaxPageSize {
+		wh.GetThrottledLogger().Warn("GetHistory page size is larger than threshold",
+			tag.WorkflowID(getRequest.Execution.GetWorkflowId()),
+			tag.WorkflowRunID(getRequest.Execution.GetRunId()),
+			tag.WorkflowDomainID(domainID), tag.WorkflowSize(int64(getRequest.GetMaximumPageSize())))
+		getRequest.MaximumPageSize = common.Int32Ptr(common.GetHistoryMaxPageSize)
+	}
+
+	execution := getRequest.Execution
+	token := &getHistoryContinuationToken{}
+
+	var runID string
+	var nextEventID int64
+	var isWorkflowRunning bool
+
+	// this function return the following 5 things,
+	// 1. the workflow run ID
+	// 2. the last first event ID (the event ID of the last batch of events in the history)
+	// 3. the next event ID
+	// 5. error if any
```

Comment:
Please add comment for the fourth result.

Reply:
fixed.